src/Test.EndToEnd.Baselining/Program.cs

```diff
@@ -0,0 +1,123 @@
+    ///    - Load each log in order
+    ///    - Replace the GUID with a 'RID' (LogIndex + ' ' + ResultIndex)
+    ///    - Baseline the current log to the Baseline-so-far
+    ///    - Write detail report lines about how each Result was baselined
+    ///    - Save the output as the Baseline-so-far
+    ///    - Compare the output to the same file in 'Expected\', if present
+    ///    - Write 'enriched' logs to Output_Debug\ and Expected_Debug\ for human investigation.
+    ///  
+    ///    
+    ///  Usage
+    ///  =====
+    ///   Get Test Content:
+    ///   git clone https://github.com/microsoft/sarif-sdk-test-content
+    ///   
+    ///   Run a Full Pass:
+    ///   "Test.EndToEnd.Baselining run C:\Code\sarif-sdk-test-content\BaselineE2E"
+    ///   
+    ///   Debug matching for Result 004 in log 001 in series "Spam\v1\WorkItem\mseng":
+    ///   To quickly debug a Result, copy and paste the first line from the Series log for the Series path and the RID (ex: 001 004) from the Result to debug.
+    ///   "Test.EndToEnd.Baselining debug C:\Code\sarif-sdk-test-content\BaselineE2E Spam\v1\WorkItem\mseng 001 004"
+    ///   
+    ///   Re-generate debug logs with the details as coded in BaseliningDetailEnricher:
+    ///   "Test.EndToEnd.Baselining rebuild-debug-logs C:\Code\sarif-sdk-test-content\BaselineE2E"
+    /// </summary>
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            try
+            {
+                Program program = new Program();
+
+                return Parser.Default.ParseArguments<RunOptions, DebugOptions, RebuildDebugLogsOptions>(args).MapResult(
+                    (RunOptions options) => Program.Run(options),
+                    (DebugOptions options) => Program.Debug(options),
+                    (RebuildDebugLogsOptions options) => Program.CreateDebugLogs(options),
+                    errs => 1);
+            }
+            catch (Exception ex) when (!Debugger.IsAttached)
+            {
+                Console.WriteLine(ex.ToString());
+                return 2;
+            }
+        }
+
+        static int Run(RunOptions options)
+        {
+            BaseliningTester tester = new BaseliningTester();
+            BaseliningSummary overallSummary = tester.RunAll(options.TestRootPath);
+            Console.WriteLine($"{overallSummary}");
+
+            return 0;
+        }
+
+        static int Debug(DebugOptions options)
+        {
+            BaseliningTester tester = new BaseliningTester();
+            tester.RunSeries(Path.Combine(options.TestRootPath, BaseliningTester.InputFolderName, options.DebugSeriesPath), options.DebugLogIndex, options.DebugResultIndex);
+
+            return 0;
+        }
+
+        static int CreateDebugLogs(RebuildDebugLogsOptions options)
+        {
+            BaseliningTester tester = new BaseliningTester();
+            tester.EnrichUnder(Path.Combine(options.TestRootPath, BaseliningTester.InputFolderName));
```

Comment:
>Combine [](start = 36, length = 7)

Just to emphasize the previous comment: here again, `Program` needs to know more about how the tester works than it should.